src/cloud/social/alias/randombytes.js

```diff
@@ -1,33 +1,64 @@
-// Alternative to crypto-browserify's randombytes shim which
-// falls back to Math.random when crypto.getRandomValues is
+// Alternative to crypto-browserify's randombytes shim
+// https://github.com/crypto-browserify/randombytes
+// Falls back to freedom.js core.crypto when crypto.getRandomValues is
 // unavailable, which is currently the case in Firefox web workers.
-// TODO: Fallback to something better such as freedomjs' core.crypto
-//       module (difficult because it is an asynchronous API).
+// TODO: post FF48 (2016-08-02) remove fill and just use built-in crypto
 
-// Firefox raises an error if we so much as even try
-// to *access* crypto, hence this bizarre construction.
-var cryptoAvailable = true;
-try {
-  if (crypto) {}
-} catch(e) {
-  console.warn('crypto unavailable, falling back to Math.random');
-  cryptoAvailable = false;
+const cryptoAvailable = typeof crypto !== 'undefined';
+var getBufferedRandomBytes, refreshBuffer, offset;
+if (!cryptoAvailable) {
+  // Filling with freedom.js core.crypto, which requires a buffer due to async
+  var rand = freedom['core.crypto'](),
+      buf;
+  refreshBuffer = function(size) {
+    return rand.getRandomBytes(size).then(function (bytes) {
+      buf = new Uint8Array(bytes);
+      offset = 0;
+    }, function (err) {
+      console.log(err);
+    });
+  }.bind(this);
+  refreshBuffer(10000);  // initial randomness
+  offset = 0;
+
+  getBufferedRandomBytes = function(buffer) {
+    if (buffer.buffer) {
+      buffer = buffer.buffer;
+    }
+    var size = buffer.byteLength,
+        view = new Uint8Array(buffer),
+        i;
+    if (offset + size > buf.length) {
+      throw new Error('Insufficient Randomness Allocated.');
+    }
+    for (i = 0; i < size; i += 1) {
```

Comment:
Agreed, slight risk that the shallowness could come to bite us but I'll give it a go.